assets/js/src/settings/pages/key_activation/index.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useContext } from 'react';
+import MailPoet from 'mailpoet';
+import { useSelector, useAction, useSetting } from 'settings/store/hooks';
+import { GlobalContext } from 'context';
+import { t } from 'common/functions';
+import { MssStatus } from 'settings/store/types';
+import { Label, Inputs } from 'settings/components';
+import SetFromAddressModal from 'common/set_from_address_modal';
+import { KeyMessages, MssMessages, PremiumMessages } from './messages';
+
+export default function KeyActivation() {
+  const { notices } = useContext<any>(GlobalContext);
+  const state = useSelector('getKeyActivationState')();
+  const setState = useAction('updateKeyActivationState');
+  const verifyMssKey = useAction('verifyMssKey');
+  const verifyPremiumKey = useAction('verifyPremiumKey');
+  const installPremiumPlugin = useAction('installPremiumPlugin');
+  const activatePremiumPlugin = useAction('activatePremiumPlugin');
+  const [senderAddress] = useSetting('sender', 'address');
+  const [unauthorizedAddresses] = useSetting('authorized_emails_addresses_check');
+
+  const verifyKey = async (event) => {
+    if (!state.key) {
+      notices.error(<p>{t('premiumTabNoKeyNotice')}</p>, { scroll: true });
+      return;
+    }
+    const isTrusted = event.isTrusted;
```

Comment:
This had a [better name](https://github.com/mailpoet/mailpoet/blob/master/assets/js/src/old_settings/premium_tab/premium_tab.jsx#L193) before, now we're losing the information on why it was used.

But - is it still relevant? It was clicked on by another settings tab. Will it be still clickable now? See https://github.com/mailpoet/mailpoet/blob/42fc6c0e3e7840da22b45727aeef213e4159f6db/views/old_settings.html#L121.

In any case we must not forget to remove this when "send with" tab is ready!

Reply:
This variable was already there and is passed to the `verifyMssKey` action. I just reused it for the modal. I will rename it to `isUserTriggered`